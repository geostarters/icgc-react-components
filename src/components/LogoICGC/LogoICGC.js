//@flow

import React from "react";
import PropTypes from "prop-types";
// eslint-disable-next-line no-unused-vars
import styles from "./LogoICGC.css";

export default function LogoICGC(props) {

	const logoProps = {
		addText: false,
		position: "bottom-right",
		url: "http://www.icgc.cat/",
		logoType: "icgc-logo",
		...props
	};

	return (
		<div className={logoProps.position} >
			<a className={logoProps.logoType} target="_blank" rel="noopener noreferrer" href={logoProps.url} aria-label="Icgc" ></a>
			{logoProps.addText &&  <div className="logo-text">{logoProps.addText}</div>}
		</div>
	);

}

LogoICGC.propTypes = {
	position: PropTypes.string,
	addText: PropTypes.string,
	url: PropTypes.string,
	logoType: PropTypes.string
};
