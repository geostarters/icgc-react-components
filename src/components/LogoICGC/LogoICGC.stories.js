//@flow

import React from "react";

import LogoICGC from "./LogoICGC";

import "semantic-ui-css/semantic.min.css";

export default {
	title: "Logo ICGC",
	component: LogoICGC
};

const Template = args => (<LogoICGC {...args}/>);

export const _LogoICGC = Template.bind({});
_LogoICGC.storyName = "Logo ICGC";

export const LogoIcgcPosition = Template.bind({});
LogoIcgcPosition.storyName = "Logo ICGC bottom-right";
LogoIcgcPosition.args = {
	position: "bottom-right"
};

export const LogoIcgcVersioAmbText = Template.bind({});
LogoIcgcVersioAmbText.storyName = "Logo ICGC versió amb text";
LogoIcgcVersioAmbText.args = {
	logoType: "icgc-logotxt"
};

export const LogoIcgcVersio2AmbText = Template.bind({});
LogoIcgcVersio2AmbText.storyName = "Logo ICGC versió 2 amb text";
LogoIcgcVersio2AmbText.args = {
	logoType: "icgc-logo2-text"
};

export const LogoIcgcVersio2BlancAmbText = args => (
	<div style={{borderBottom: "2px solid gray", width: "100%", height: "200px", backgroundColor: "gray", padding: "10px", position: "relative"}}>
		<LogoICGC {...args}/>
	</div>
);
LogoIcgcVersio2BlancAmbText.storyName = "Logo ICGC versió 2 blanc amb text";
LogoIcgcVersio2BlancAmbText.args = {
	logoType: "icgc-logo-white-text",
	position: "top-left"
};

export const LogoIcgcAmbTextPropi = Template.bind({});
LogoIcgcAmbTextPropi.storyName = "Logo ICGC amb text propi";
LogoIcgcAmbTextPropi.args = {
	addText: "ICGC",
	position: "top-right"
};
