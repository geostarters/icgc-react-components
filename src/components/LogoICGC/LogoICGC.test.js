// @flow

import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import LogoICGC from "./LogoICGC";


describe("LogoICGC component", () =>{

	it("matches the snapshot ", ()=>{

		const component = shallow(<LogoICGC></LogoICGC>);
		expect(toJSON(component)).toMatchSnapshot();

	});

	it("has logotype = icgc-logotxt", ()=>{

		const component = shallow(<LogoICGC logoType="icgc-logotxt"></LogoICGC>);
		expect(component.find("a").hasClass("icgc-logotxt")).toEqual(true);

	});

});

