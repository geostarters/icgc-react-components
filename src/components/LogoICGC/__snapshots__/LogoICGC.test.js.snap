// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoICGC component matches the snapshot  1`] = `
<div
  className="bottom-right"
>
  <a
    aria-label="Icgc"
    className="icgc-logo"
    href="http://www.icgc.cat/"
    rel="noopener noreferrer"
    target="_blank"
  />
</div>
`;
